Automattic/wp-calypso
# Site Settings: Move Writing section to a separate chunk

We've been working a lot on the Site Settings recently. A lot of stuff has been introduced, enhanced or improved, but this also has some cons: the `settings` chunk is now 1.12MB 😱 

![](https://cldup.com/M1UdN8a_9R.png)

It is obvious that we need to split `settings` to smaller chunks. This has also been discussed in p4TIVU-5Zs-p2 and recommended by @dmsnell and @mtias.

So this PR moves the Writing settings section to a separate chunk. This is just one of the PRs that take care of this - there is also one that suggests moving the General section to a separate chunk - #12498.

This is not introducing any functional or behavioral changes. But since it's affecting all of the routes and functionality that resides within the Writing settings section, extensive testing of the Writing section and its settings will be required.

To test:
* Checkout this branch, or get it going on calypso.live
* Go to `/settings/writing/$site`, where `$site` is one of your WordPress.com sites.
* Verify the tab loads correctly with no errors/warnings.
* Change to another tab and come back to Writing, verify it works like before.
* Play with all settings in the Writing section and verify they are saved and retrieved correctly.
* Click on the "Categories" button and verify categories management works properly.
* Go to `/settings/writing/$site`, where `$site` is one of your connected Jetpack sites.
* Verify the tab loads correctly with no errors/warnings.
* Change to another tab and come back to Writing, verify it works like before.
* Play with all settings in the Writing section and verify they are saved and retrieved correctly.
* Click on the "Categories" button and verify categories management works properly.
